### PR TITLE
RadosGW API: incorrect bucket quota in response to HEAD /{bucket}/?usage

### PR DESCRIPTION
**Pull request by shreyanshjain7174**
When we try to get the bucket usage via various methods, through curl or while accessing rgw api endpoint at HEAD /{bucket}/?usage doesn't return the updated information. The endpoint was always returning the user quota and not the actual bucket quota which we see after querying the endpoint.

Fixes: https://tracker.ceph.com/issues/62737
Signed-off-by: shreyanshjain7174 <ssanchet@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/rgw/rgw_rest_s3.cc [hunk 1]
```diff
@@ -2345,11 +2345,12 @@ static void dump_bucket_metadata(req_state *s, rgw::sal::Bucket* bucket)
   // only bucket's owner is allowed to get the quota settings of the account
   if (bucket->is_owner(s->user.get())) {
     auto user_info = s->user->get_info();
+    auto bucket_info = s->bucket->get_info().quota; // bucket quota
```
Comment (dang) [inline src/rgw/rgw_rest_s3.cc:2348]: This is the bucket quota, not the bucket info.  Either it should be named `bucket_quota` or it should store the info, like `user_info` does.

File: src/rgw/rgw_sal.h [hunk 2]
```diff
@@ -791,7 +791,8 @@ class Bucket {
     /* dang - This is temporary, until the API is completed */
     virtual rgw_bucket& get_key() = 0;
     virtual RGWBucketInfo& get_info() = 0;
-
+    //virtual RGWQuotaInfo& bucket_quota() = 0;
+
```
Comment (dang) [inline src/rgw/rgw_sal.h:795]: This seems unnecessary.

**Review by dang — DISMISSED** [review]

File: src/rgw/rgw_sal.h [hunk 3]
```diff
@@ -791,7 +791,7 @@ class Bucket {
     /* dang - This is temporary, until the API is completed */
     virtual rgw_bucket& get_key() = 0;
     virtual RGWBucketInfo& get_info() = 0;
-
+
```
Comment (alimaredia) [inline src/rgw/rgw_sal.h:794]: This tab addition should still be removed from this PR

File: src/rgw/rgw_rest_s3.cc (hunk 1 above)
Comment (shreyanshjain7174) [reply src/rgw/rgw_rest_s3.cc]: Done!

File: src/rgw/rgw_sal.h (hunk 3 above)
Comment (shreyanshjain7174) [reply src/rgw/rgw_sal.h]: done!

**Review by alimaredia — APPROVED** [review]

**Pull request merged**